lib/controller-inputs/AnalogInputGroup.cpp

```diff
@@ -0,0 +1,66 @@
+#include "AnalogInputGroup.h"
+
+AnalogInputGroup::AnalogInputGroup(AnalogBusIn& inputs, uint8_t numPins, uint8_t numSampleToAverage)
+    : m_inputs(inputs), m_numPins(numPins), m_num_samples_to_average(numSampleToAverage) {}
+
+uint8_t AnalogInputGroup::getNumInputs() const {
+  return m_numPins;
+}
+
+void AnalogInputGroup::read() {
```

Comment:
Good point! We prob need to test for the decay factor though, I'll set it to 0.8 arbitrarily for now...

Reply:
On second thought, I wonder what is the performance difference? I think using EWMA would make slow down each read as we are using float multiplications...and this would make fewer polls in one interval...it would make it faster for getValues though since it can be a straight copy...
What about we make the averaging algorithm configurable, as what was done for debouncing for digitalInputsGroup

Reply:
EWMA should be very fast on our processor (Cortez M4 with FPU @ 180 MHz), I would expect quicker than the multiple array accesses + operations & if statements required for moving window. If you are curious you could measure the performance difference.

Reply:
Decent way to estimate a good decay factor is generate some dummy data and apply the filter in a spreadsheet w/ graph, then look at response in graph after tweaking. 0.8 is probably good for now.